### ISSUE TITLE
API/usage questions

### ISSUE THREAD

**Issue by deniederhut** [ORNL/tx2#3]
- The wrapper class has a separate prepare method -- is there a reason this work isn't done during init?
- Is there a reason we're requiring dataframes as input and not allowing numpy arrays?
- Super happy to see type annotations here. That's not a question.
- What happens if I don't have a test set, only some training data?
- Different operating systems have different conventions for disk cache locations (e.g. ~/.cache). Is this something you've thought about?
- Can I use a different embedding than UMAP? E.g. tSNE?
- Would it make sense to default to iloc[:, 0] when no column name is provided for the input?

**Comment by WildfireXIII** [comment]
Just an update, we've changed the wrapper to accept either numpy arrays or pandas series for train/test text/labels, rather than expecting dataframes and column names. I think this addresses the most important points you made.
